packages/runtime/src/copilotinstructions.ts

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { WorkspaceFileSystem, WorkspaceFile } from "@genaiscript/core";
+import { frontmatterTryParse, isGlobMatch } from "@genaiscript/core";
+
+/**
+ * Represents a copilot instruction file with its content and metadata
+ */
+export interface CopilotInstruction {
+  /** The file path of the instruction file */
+  filename: string;
+  /** The instruction content */
+  content: string;
+  /** Parsed frontmatter metadata */
+  metadata?: {
+    /** File patterns this instruction applies to */
+    applyTo?: string | string[];
+    /** Description of the instruction */
+    description?: string;
+    [key: string]: any;
+  };
+}
+
+/**
+ * Configuration options for importing copilot instructions
+ */
+export interface CopilotInstructionsOptions {
+  /** Whether to include general copilot instructions file (default: true) */
+  includeGeneral?: boolean;
+  /** Custom paths to search for instruction files (default: [".github/instructions", ".github"]) */
+  instructionPaths?: string[];
+  /** Custom patterns to match instruction files (default: ["*.instructions.md", "copilot-instructions.md"]) */
+  instructionPatterns?: string[];
+}
+
+/**
+ * Runtime helper to automatically import copilot instruction files based on env.files.
+ *
+ * This function searches for GitHub Copilot instruction files and filters them based on
+ * file patterns specified in their frontmatter `applyTo` field, matching against the
+ * files provided in env.files.
+ *
+ * @param workspace - The workspace file system to read files from
+ * @param envFiles - Array of files from env.files to match against instruction patterns
+ * @param options - Configuration options for the import
+ * @returns Promise that resolves to an array of relevant copilot instructions
+ *
+ * @example
+ * ```typescript
+ * // Import instructions that apply to current env.files
+ * const instructions = await importCopilotInstructions(workspace, env.files);
+ *
+ * // Use the instructions in your script
+ * for (const instruction of instructions) {
+ *   console.log(`Applying instruction from ${instruction.filename}:`);
+ *   console.log(instruction.content);
+ * }
+ * ```
+ */
+export async function importCopilotInstructions(
+  workspace: WorkspaceFileSystem,
+  envFiles: WorkspaceFile[] | string[],
+  options: CopilotInstructionsOptions = {},
```

Comment:
Add optional chat generation context and import the instructions as system prompts in that context.

See other runtime helpers for pattern.